python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -0,0 +1,620 @@
+                                " please read the cuML documentation for"
+                                " more information")
+        super(RandomForestRegressor, self).__init__(handle, verbose)
+
+        self.split_algo = split_algo
+        self.criterion = split_criterion
+        self.split_criterion = split_criterion
+        self.min_rows_per_node = min_rows_per_node
+        self.bootstrap_features = bootstrap_features
+        self.rows_sample = rows_sample
+        self.max_leaves = max_leaves
+        self.n_estimators = n_estimators
+        self.max_depth = max_depth
+        self.max_features = max_features
+        self.bootstrap = bootstrap
+        self.verbose = verbose
+        self.n_bins = n_bins
+        self.n_cols = None
+        self.accuracy_metric = accuracy_metric
+        self.quantile_per_tree = quantile_per_tree
+
+        cdef RandomForestMetaData[float, float] *rf_forest = \
+            new RandomForestMetaData[float, float]()
+        self.rf_forest = <size_t> rf_forest
+        cdef RandomForestMetaData[double, double] *rf_forest64 = \
+            new RandomForestMetaData[double, double]()
+        self.rf_forest64 = <size_t> rf_forest64
+    """
+    TODO:
+        Add the preprocess and postprocess functions
+        in the cython code to normalize the labels
+    """
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        del state['handle']
+
+        cdef size_t params_t = <size_t> self.rf_forest
+        cdef  RandomForestMetaData[float, float] *rf_forest = \
+            <RandomForestMetaData[float, float]*>params_t
+
+        cdef size_t params_t64 = <size_t> self.rf_forest64
+        cdef  RandomForestMetaData[double, double] *rf_forest64 = \
+            <RandomForestMetaData[double, double]*>params_t64
+
+        state['verbose'] = self.verbose
+        # state["trees"] = <void *> rf_forest.trees
+
+        if self.dtype == np.float32:
+            print(" dtype is float32")
+            state["rf_params"] = rf_forest.rf_params
+            del state["rf_forest"]
+        else:
+            print(" dtype is float64")
+            state["rf_params64"] = rf_forest64.rf_params
+            del state["rf_forest64"]
+
+        return state
+
+    def __del__(self):
+        cdef RandomForestMetaData[float, float]* rf_forest = \
+            <RandomForestMetaData[float, float]*><size_t> self.rf_forest
+        cdef RandomForestMetaData[double, double]* rf_forest64 = \
+            <RandomForestMetaData[double, double]*><size_t> self.rf_forest64
+        free(rf_forest)
+        free(rf_forest64)
+
+    def __setstate__(self, state):
+        super(RandomForestRegressor, self).__init__(handle=None,
+                                                    verbose=state['verbose'])
+        cdef  RandomForestMetaData[float, float] *rf_forest = \
+            new RandomForestMetaData[float, float]()
+
+        # rf_forest.trees = state["trees"]
```

Comment:
```suggestion
```

I presume the commented lines are not needed, right?

Reply:
yes, i will remove them